uberenv.py

```diff
@@ -864,13 +900,48 @@ def patch(self):
     def create_spack_env(self):
         # Create Spack Environment
         print("[creating spack env]")
+        if self.spack_env_file is None:
+            self.spack_env_file = ""
         spack_create_cmd = "{0} env create -d {1} {2}".format(self.spack_exe(use_spack_env=False),
             self.spack_env_directory, self.spack_env_file)
         sexe(spack_create_cmd, echo=True)
+        
+        # Find pre-installed compilers and packages and stop uberenv.py
+        if self.spack_setup_environment:
+            # Finding compilers
+            print("[finding compilers]")
+            if self.spack_compiler_paths is None:
+                spack_compiler_find_cmd = "{0} compiler find".format(self.spack_exe())
+            else:
+                spack_compiler_find_cmd = "{0} compiler find {1}".format(self.spack_exe(), self.spack_compiler_paths)
```

Comment:
Meta-question (outside the scope of this PR):

Should we have a `spack_command(command, additional_args)` function in uberenv?
That way, this line could be written as something like:
```python
    spack_compiler_find_cmd = spack_command("compiler find", self.spack_compiler_paths)
```